### ISSUE TITLE
Minced version issue 

### ISSUE THREAD

**Issue by emmadoughty** [tseemann/prokka#396]
I've been using prokka for a few weeks without issue on a few servers but in the last week or so found that when I run prokka on any of these servers, it exits with an error message (below).

Error: A JNI error has occurred, please check your installation and try again
Exception in thread "main" java.lang.UnsupportedClassVersionError: minced has been compiled by a more recent version of the Java Runtime (class file version 55.0), this version of the Java Runtime only recognizes class file versions up to 52.0
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:763)
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:467)
	at java.net.URLClassLoader.access$100(URLClassLoader.java:73)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:368)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:362)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:361)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at sun.launcher.LauncherHelper.checkAndLoadMain(LauncherHelper.java:495)
[09:21:05] Could not determine version of minced - please install version 2.0 or higher

Prokka is installed with conda on these servers, sometimes with a single user conda installation, sometimes with a conda installation for multiple users though only I (admin) can install things with conda on the servers with multiple users. I'm not aware that I have made any changes to the prokka installation or dependencies (the versions for everything in each conda env is pinned). Uninstalling/reinstalling prokka has not worked, neither has installing prokka in a new conda env on its own. I also can't run minced or update it (this would have been installed with conda as a dependency for prokka).

I'm really new to all sys admin type stuff and a little lost. All/any help would be appreciated!

**Comment by ALFLAG** [comment]
Sorry Emma,
I won't be able to help you.
I add a comment just to say that I have the same problem. I also installed the tool with conda too (from bioconda channel).

**Comment by dutchscientist** [comment]
I just reinstalled Prokka due to Conda issues (and tbl2asn expiring), and can confirm this problem with 1.13.7. Somehow it goes wrong with minced.

**Comment by tseemann** [comment]
This error is a difference between the version of Java that the mincer.jar file was compiled with, and the version of Java that is trying to run it.

TL;DR  your `java -version` is too old.  not sure what the bioconda people have specified for prokka.

THe problem is that I think `snpEff.jar` might only work on the OLD java. So we have a problem.

**Comment by dutchscientist** [comment]
Thanks, Torsten. At least there is a work-around for now,

**Comment by emmadoughty** [comment]
Thanks for looking into this, Torsten.  I played around with this again and found that specifying the version of prokka as 1.13.7 resolves this. Conda was installing v1.12 by default. I ran the installation as below:

conda create -n <myenv> -c conda-forge -c bioconda -c defaults prokka=1.13.7-0

It does still offer these warnings on installation (though runs fine and I can't see any issues with the outputs):

Executing transaction: | WARNING conda.gateways.disk.delete:unlink_or_rename_to_trash(140): Could not remove or rename /usr/local/software/conda/envs/GCP/lib/5.26.2/x86_64-linux-thread-multi/perllocal.pod.  Please remove this file manually (you may need to reboot to free file handles)
WARNING conda.gateways.disk.delete:unlink_or_rename_to_trash(140): Could not remove or rename /usr/local/software/conda/envs/GCP/lib/5.26.2/x86_64-linux-thread-multi/perllocal.pod.  Please remove this file manually (you may need to reboot to free file handles)
/ WARNING conda.gateways.disk.delete:unlink_or_rename_to_trash(140): Could not remove or rename /usr/local/software/conda/envs/GCP/lib/5.26.2/x86_64-linux-thread-multi/perllocal.pod.  Please remove this file manually (you may need to reboot to free file handles)

**Comment by rjcie** [comment]
Hi, 

I did notice the same conflict with 'minced' after re-installing Prokka recently (through Conda).
The solution that worked for me was download the current version of 'minced' from 'github' and use them to replace the version of 'minced' installed through Conda.

Doing this I could run Prokka without any issue or warning.
Hope this is useful for anyone!

Here is how I did it:

```shell
##################      INSTALLATION INSTRUCTIONS   ############################

# (1) Install Prokka through Conda
conda install -c bioconda prokka

# (2) Update the installed version of minced
#     Updating minced will remove the following error
#######         ERROR            ############
# [00:20:26] Looking for 'minced' - found /home/rjcieza/miniconda3/bin/minced
# Error: A JNI error has occurred, please check your installation and try again
#############################################

# My prokka installation was through Miniconda, so I will look for minced in the Miniconda DIR
# Minced is intalled in 2 DIR inside Miniconda: [ bin ] [ pkgs ]

# (2.1) Remove minced from DIR [ bin ]
cd ~/miniconda3/bin/
rm minced*

# (2.2) Remove minced from DIR [ pkgs ]
cd pkgs/minced-0.4.0-0/bin/
rm minced*
cd pkgs/minced-0.4.0-0/share/minced-0.4.0-0/
rm minced*

# (2.3) Get the latest version of minced
cd ~
git clone https://github.com/ctSkennerton/minced.git
cd minced/
make
# Confirm minced can be executed
# Always keep minced and minced.jar in the same folder!
java -jar minced.jar --help

# (2.4) Copy minced & minced.jar into the directories where minced was removed from
cp minced* ~/miniconda3/bin/
cp minced* ~/miniconda3/pkgs/minced-0.4.0-0/bin/
cp minced* ~/miniconda3/pkgs/minced-0.4.0-0/share/minced-0.4.0-0/ 

# (2.5) Remove minced DIR outside of miniconda binaries
rm -r ~/minced/
```

**Comment by dana33335446** [comment]
I used brew install in MACOS and solved the problem using the method @dutchscientist mentioned.

Download minced and minced.jar from https://github.com/ctSkennerton/minced/releases, mv minced /usr/local/bin/ & mv minced.jar /usr/local/bin/, and it was done.

**Comment by sipesk** [comment]
Im having a similar problem on MacOS, Ventura 13.6;

Installed prokka with `conda install -c bioconda prokka`

[11:19:49] Looking for 'minced' - found /usr/local/bin/minced
Exception in thread "main" java.lang.UnsupportedClassVersionError: minced has been compiled by a more recent version of the Java Runtime (class file version 58.0), this version of the Java Runtime only recognizes class file versions up to 52.0
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:756)
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:473)
	at java.net.URLClassLoader.access$100(URLClassLoader.java:74)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:369)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:363)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:362)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:355)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
	at sun.launcher.LauncherHelper.checkAndLoadMain(LauncherHelper.java:632)
[11:19:49] Could not determine version of minced - please install version 2.0 or higher


`$minced -V`
Exception in thread "main" java.lang.UnsupportedClassVersionError: minced has been compiled by a more recent version of the Java Runtime (class file version 58.0), this version of the Java Runtime only recognizes class file versions up to 52.0
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:756)
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:473)
	at java.net.URLClassLoader.access$100(URLClassLoader.java:74)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:369)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:363)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:362)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:355)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
	at sun.launcher.LauncherHelper.checkAndLoadMain(LauncherHelper.java:632)



`$conda install minced`
Collecting package metadata (current_repodata.json): done
Solving environment: done

# All requested packages already installed.


Can someone advise?

**Comment by dutchscientist** [comment]
Have you installed Prokka in a fresh Conda environment, without conflicting packages? The Java runtime is often a problem when having multiple packages in a single environment. I had problems with Prokka when installing it in a Python 3.10 environment, but it was happy in a Python 3.7 environment.

mamba create -n prokka python=3.7 --yes
(mamba is a lot quicker than conda!)
And then install.
